homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -380,82 +395,78 @@ def _build_observations_by_template(self):
                 continue
 
-            template = obs.get(CONF_VALUE_TEMPLATE)
-            observations_by_template.setdefault(template, []).append(obs)
+            template = observation.value_template
+            observations_by_template.setdefault(template, []).append(observation)
 
         return observations_by_template
 
     def _process_numeric_state(self, entity_observation):
         """Return True if numeric condition is met, return False if not, return None otherwise."""
-        entity = entity_observation["entity_id"]
+        entity = entity_observation.entity_id
 
         try:
             if condition.state(self.hass, entity, [STATE_UNKNOWN, STATE_UNAVAILABLE]):
                 return None
             return condition.async_numeric_state(
                 self.hass,
                 entity,
-                entity_observation.get("below"),
-                entity_observation.get("above"),
+                entity_observation.below,
+                entity_observation.above,
                 None,
-                entity_observation,
+                entity_observation.to_dict(),
             )
         except ConditionError:
             return None
 
     def _process_state(self, entity_observation):
-        """Return True if state conditions are met."""
-        entity = entity_observation["entity_id"]
+        """Return True if state conditions are met, return False if they are not. Returns None if the state is unavailable."""
+        entity = entity_observation.entity_id
 
         try:
             if condition.state(self.hass, entity, [STATE_UNKNOWN, STATE_UNAVAILABLE]):
                 return None
 
-            return condition.state(
-                self.hass, entity, entity_observation.get("to_state")
-            )
+            return condition.state(self.hass, entity, entity_observation.to_state)
         except ConditionError:
             return None
 
     def _process_multi_state(self, entity_observation):
-        """Return True if state conditions are met."""
-        entity = entity_observation["entity_id"]
+        """Return True if state conditions are met, otherwise return None. Never return False as all other states should have their own probabilities configured."""
+        entity = entity_observation.entity_id
 
         try:
-            if condition.state(self.hass, entity, entity_observation.get("to_state")):
+            if condition.state(self.hass, entity, entity_observation.to_state):
                 return True
         except ConditionError:
             return None
+        return None
 
     @property
     def extra_state_attributes(self):
         """Return the state attributes of the sensor."""
         attr_observations_list = [
-            obs.copy() for obs in self.current_observations.values() if obs is not None
+            obs.to_dict()
+            for obs in self.current_observations.values()
+            if obs is not None
```

Comment:
Please use a full word here:
```suggestion
            if observation is not None
```